File: functional-interface-replacement/src/main/java/io/functional/patterns/oo/functionalinterface/PersonFirstNameSort.java
package io.functional.patterns.oo.functionalinterface;

import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.List;

public class PersonFirstNameSort {

	public static void main(String[] args) {
		Person p1 = new Person ("jack", 30);
		Person p2 = new Person ("Peter", 23);
		Person p3 = new Person ("Anabele", 18);
		
		List<Person> people = Arrays.asList(p1, p2, p3);
		Collections.sort(people, new Comparator<Person>() {
			public int compare(Person p1, Person p2) {
				return p1.getName().compareTo(p2.getName());
			}
		});
			
		
	}
	
	
}

File: functional-interface-replacement/src/main/scala/io/functional/patterns/functionalinterface/PersonFirstNameSort.scala
package io.functional.patterns.functionalinterface

object PersonFirstNameSorter extends App { 
  val p1 = Person("Jack", 38)
  val p2 = Person("Peter", 23);
  val p3 = Person("Anabele", 18);
  
  
  val people = Vector(p3, p1, p2)
  
  people.sortWith((p1, p2) => p1.name < p2.name)
  
}


case class Person(name: String, age: Int)